### DISCUSSION TITLE
Textbox overflows the boundary in group on windows

### DISCUSSION THREAD

**Discussion by umtdemr** [fabricjs/fabric.js#8791] [Q&A]
Hi, I made a sticky note shape. That is a group and it has two shapes. One is a rect and the other one is a textbox. And we can edit the textbox while double-clicking on the group. 

I want to fit the text into the group while the user typing. For that, I calculate the proper font size according to the text height and apply it to the editing textbox. Everything is good so far but I noticed this; It works properly on _macOS (Ventura) chrome Version 111.0.5563.64 (Official Build) (arm64)_ but the text sometimes overflows on _windows chrome 110.0.5481.77 (Official Build) (64 bit)_.

Here are the examples from both os (the red area shows the boundary that textbox has):

Macos
![sticky note on macos chrome](https://user-images.githubusercontent.com/61540809/226330102-827da269-871a-45a0-af5b-74f1d26fd7da.png)

Windows
![sticky note on windows chrome](https://user-images.githubusercontent.com/61540809/226330122-f98d98c5-b202-49a7-9c49-cb80ee6d5c37.jpeg)

If you want to try it: [https://jsfiddle.net/mediumgoal/et2p0nwb/](https://jsfiddle.net/mediumgoal/et2p0nwb/)
This is the text I tried with: simply dummy text in a sticky note fthe wight is higher this happens text sometimes overflows the boundray it hgas and caret position renders wrong I'm not sure why but we will se this you see this we are using the same updated values deneme denemee deneme eneme deneme deneme deneme deneme deneme deneme deneme denemeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee

Do you have any idea why this is happening?

**Comment by ShaMan123** [comment]
I tried to understand your repro but couldn't w/o diving in, something I don't want to do.
Can you make it simpler?

**Reply by umtdemr** [reply]
@ShaMan123 sorry for the late reply. Actually, you don't need to understand the whole code. Let me explain it.

Line 8 - 95: Because I don't want the text to be overflowed according to splitBygrapheme (it breaks the text in the middle of the word),  I override fabric js' default _wrapLine function to overflow text like CSS's word-break. 

Line 98 - 271: Here, I try to find the proper font size with the given width and text for the textbox. This is for avoiding overflow in y axis.

Line 274 - 408: I create a new textbox by double-clicking. This is for editing text.
   Line 354 - 406: In here, I listen to the textbox's changed event and I apply the proper font size to the textbox.

Line 411 - 460: I create the sticky note group with a rectangle and textbox.
